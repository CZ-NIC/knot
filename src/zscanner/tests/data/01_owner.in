$ORIGIN	.
$TTL	1

; OK
.		NS	@	; The simplest owner
tld.		NS	@	; FQD tld owner
tld		NS	@	; Relative form
		NS	@	; The previous owner
*.		NS	@	; FQD with asterisk
*		NS	@	; Alone asterisk
*.*		NS	@	; More asterisks
*a.a*a.**	NS	@	; Also possible
@		NS	@	; Use origin
0123456789	NS	@	; Digits
0/25.2.0.192.in-addr.arpa.	NS	@	; CIDR notation
_a_.-b-c-./d/.	NS	@	; Allowed characters '_' '-' '/' anywhere
ABCDEFGHIJKLMNOPQRSTUVWXYZ	NS	@	; All upper-case letters
abcdefghijklmnopqrstuvwxyz	NS	@	; All lower-case letters
\000\0320\ \\\"\.\@\*.tld.	NS	@	; Label with special chars
b.a.9.8.7.6.5.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa.	NS	@	; IPv6 reverse
12345678901234567890123456789012345678901234567890123456789012\051.tld.		NS	@	; Label of maximal length
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.1234567890123456789012345678901234567890123456789012345678901.	NS	@	; Domain name of maximal length
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.1234567890123456789012345678901234567890123456789012345678901	NS	@	; Domain name of maximal length (after appending origin)

; KO
&	NS	@	; Bad (unslashed) character
	NS	@	; Bad previous
.a	NS	@	; Leading dot
@@	NS	@	; Double @@
..	NS	@	; Missing label between dots
\1	NS	@	; Slash notation requires 3 digits
\12	NS	@	; Slash notation requires 3 digits
12345678901234567890123456789012345678901234567890123456789012\0514.tld.	NS	@	; Label exceeded maximal length
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.12345678901234567890123456789012345678901234567890123456789012.	NS	@	; Domain name exceeded maximal length
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.12345678901234567890123456789012345678901234567890123456789012	NS	@	; Domain name exceeded maximal length (after appending origin)
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.1234567890123456789012345678901234567890123456789012345678901.1	NS	@	; Domain name exceeded maximal length (maximal dname length check is after each valid label)
123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.123456789012345678901234567890123456789012345678901234567890123.1234567890123456789012345678901234567890123456789012345678901.123456789012345678901234567890123456789012345678901234567890123	NS	@	; Domain name exceeded maximal length (maximal dname length check is after each valid label)
