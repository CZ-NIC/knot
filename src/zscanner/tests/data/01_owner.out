OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=03746C6400
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=03746C6400
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=03746C6400
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=012A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=012A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=012A012A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=022A6103612A61022A2A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=0A3031323334353637383900
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=04302F3235013201300331393207696E2D61646472046172706100
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=035F615F052D622D632D032F642F00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=1A4142434445464748494A4B4C4D4E4F505152535455565758595A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=1A6162636465666768696A6B6C6D6E6F707172737475767778797A00
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=09002030205C222E402A03746C6400
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=0162016101390138013701360135013001300130013001300130013001300130013001300130013001300130013001300138016201640130013101300130013203697036046172706100
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=3F31323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333435363738393031323303746C6400
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=3F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333D3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303100
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
OWNER=3F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333F3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303132333D3132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930313233343536373839303100
CLASS=0001
RRTTL=00000001
RTYPE=0002
RDATA=00
------
WARNG=ZS_BAD_OWNER
------
WARNG=ZS_BAD_PREVIOUS_OWNER
------
WARNG=ZS_BAD_DNAME_CHAR
------
WARNG=ZS_BAD_DNAME_CHAR
------
WARNG=ZS_BAD_DNAME_CHAR
------
WARNG=ZS_BAD_NUMBER
------
WARNG=ZS_BAD_NUMBER
------
WARNG=ZS_LABEL_OVERFLOW
------
WARNG=ZS_DNAME_OVERFLOW
------
WARNG=ZS_DNAME_OVERFLOW
------
WARNG=ZS_DNAME_OVERFLOW
------
WARNG=ZS_DNAME_OVERFLOW
------
